### ISSUE TITLE
Unexpected errors on sync code

### ISSUE THREAD

**Issue by Zac-HD** [Zac-HD/flake8-trio#29]
```python
def sync():
    while ...:
        if ...:
            return
        yield
```

Running `flake8-trio` on this example crashed with `AttributeError: 'Visitor107_108' object has no attribute 'safe_decorator'`.  I fixed that in d484205ba17ecf88a651b0aab9db494c4a98092c, but there are still missing-checkpoint warnings - which is a false alarm for a sync function!

As well as fixing this let's add a test that for each test file, if you remove each "async " and "await " keyword and then run the linter, there are no TRIO warnings (except TRIO105 missing-await warnings).

**Comment by jakkdl** [comment]
Ah shoot, lost that in one of many refactors going on around safe_decorator. Surprised that flake8 doesn't catch that, I think I'll start running pylint (which does) against the code again - and maybe worth adding it to tox too (but needs a lot of disabled warnings
The crash also would've been caught by running the fuzzer, now that I lowered the examples I should make a habit (i.e. add to my pre-push alias) of running that before pushing.

Trio106 will also be raised on sync code. As will 100, 101, 103 & 104. Unless any of those should be changed to ignore sync code that's 6/11 checks that shouldn't have that test - so I'll add another keyword to the test file to enable/disable that.

And will fix the missing checkpoint warning and add an explicit test :+1:

**Comment by Zac-HD** [comment]
Instead of a test-file flag, let's just have a static list (in the test function) of codes which can be raised from sync code?

**Comment by Zac-HD** [comment]
Fixed by #31 and #33 🎉
